CreateSameFileAtOnceTest.testCreateSameFileAtOnce fails

The test AddMultipleFilesTest.testAddMultipleFilesSimultaneouslyTest fails because of a regression.

The test-case shows the following behavior:
* ALICE starts the session with BOB & CARL (a project with a file "dummy.sync").
* The NetworkManipulator blocks the messages of BOB & CARL
* Both create the file readme.txt
* The NetworkManipulator unblocks the messages
* This results in a session where BOB & CARL have a project with two files and ALICE has only the "dummy.sync" file.
* The test fails, because ALICE waits for the readme file (leads to a timeout exception)

I removed CARL from the test setup and the result is as before (ALICE has one file, Bob two, this leads to a timeout exception).

After removing the NetworkManipulator calls the test is succesful. I noticed that ALICE throws the exception `java.io.IOException: failed to inflate data` (Also included in the logs of the stf runs on the CI server). Therefore, I assume that ALICE is not able to receive the previously blocked message containing the corresponding FileActivity.
<details>
<summary>Exception</summary>
<code>
TRACE 16:20:32,753 [BinaryChannel-jenkins_carl_stf] (BinaryChannelConnection.java:308) processing opcode 0xFA [TFD]: id=0, chunks=1
TRACE 16:20:32,753 [BinaryChannel-jenkins_carl_stf] (BinaryChannelConnection.java:346) processing opcode 0xFB [DATA]: id=0, DATA len=273 bytes
TRACE 16:20:32,753 [saros-DispatchContext] (NetworkManipulatorImpl.java:99) intercepting incoming packet from: jenkins_carl_stf@stf_prosody_test.org/Saros
TRACE 16:20:32,753 [saros-DispatchContext] (XMPPReceiver.java:214) received binary XMPP extension: TransferDescription [elementName=ados, namespace=saros, recipient=jenkins_alice_stf@stf_prosody_test.org/Saros, sender=jenkins_carl_stf@stf_prosody_test.org/Saros, compress=true], size: 273, RX time: 0 ms [SOCKS5 (D)]
ERROR 16:20:32,753 [saros-DispatchContext] (XMPPReceiver.java:232) could not decompress extension payload
java.io.IOException: failed to inflate data
	at saros.net.internal.XMPPReceiver.inflate(XMPPReceiver.java:324)
	at saros.net.internal.XMPPReceiver.convertBinaryXMPPExtension(XMPPReceiver.java:230)
	at saros.net.internal.XMPPReceiver.access$200(XMPPReceiver.java:37)
	at saros.net.internal.XMPPReceiver$6.run(XMPPReceiver.java:169)
	at saros.util.ThreadUtils$1.run(ThreadUtils.java:34)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.zip.DataFormatException: incorrect header check
	at java.util.zip.Inflater.inflateBytes(Native Method)
	at java.util.zip.Inflater.inflate(Inflater.java:259)
	at java.util.zip.Inflater.inflate(Inflater.java:280)
	at saros.net.internal.XMPPReceiver.inflate(XMPPReceiver.java:319)
	... 9 more
</code>
</details>

I performed a manual bisect and found that:
* Test always fails with the same exception with commit 23529f7fb31f523b4ab808984ba37226ad5a82bc "[INTERNAL] Moves send logic from DTM to XMPPTransmitter". (This commit still requires an eclipse neon for stf testing)
* Test is succesful with commit b24e8ea4d267a6cff317c720e4fa92fc4060b9b1 "[UI][I] Use IntelliJ UI wrapper classes" (This commit still requires an eclipse neon for stf testing)

It seems like PR #652 introduced this issue @srossbach do you know whether this could have led to an incompatibility of the NetworkManipulator and the productive extension handling?

--- a/stf.test/build.gradle
+++ b/stf.test/build.gradle
@@ -58,16 +58,16 @@ task stfTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
     classpath = sourceSets.stfTest.runtimeClasspath
 
     useJUnit {
-        excludeCategories 'saros.stf.test.categories.FlakyTests', 'saros.stf.test.categories.FailingTests'
+        excludeCategories 'saros.stf.test.categories.FlakyTests'
     }
 }
 
-task stfFlakyAndFailingTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
+task stfFlakyTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
     testClassesDirs = sourceSets.stfTest.output.classesDirs
     classpath = sourceSets.stfTest.runtimeClasspath
     ignoreFailures = true
-    
+
     useJUnit {
-        includeCategories 'saros.stf.test.categories.FlakyTests', 'saros.stf.test.categories.FailingTests'
+        includeCategories 'saros.stf.test.categories.FlakyTests'
     }
 }

--- a/stf.test/test/saros/stf/test/categories/FailingTests.java
+++ b/stf.test/test/saros/stf/test/categories/FailingTests.java
@@ -1,9 +1,0 @@
-package saros.stf.test.categories;
-
-/**
- * Marks tests are always failing and have to be fixed! Create an issue "Fix STF failing test
- * <Testname>" before using this marker
- */
-public interface FailingTests {
-  /* category marker */
-}

--- a/stf.test/test/saros/stf/test/consistency/CreateSameFileAtOnceTest.java
+++ b/stf.test/test/saros/stf/test/consistency/CreateSameFileAtOnceTest.java
@@ -9,13 +9,11 @@ import static saros.stf.client.tester.SarosTester.CARL;
 import org.junit.After;
 import org.junit.BeforeClass;
 import org.junit.Test;
-import org.junit.experimental.categories.Category;
 import saros.stf.annotation.TestLink;
 import saros.stf.client.StfTestCase;
 import saros.stf.client.tester.AbstractTester;
 import saros.stf.client.util.Util;
 import saros.stf.shared.Constants.TypeOfCreateProject;
-import saros.stf.test.categories.FailingTests;
 
 @TestLink(id = "Saros-131_create_same_file_at_once")
 public class CreateSameFileAtOnceTest extends StfTestCase {
@@ -31,7 +29,6 @@ public class CreateSameFileAtOnceTest extends StfTestCase {
     CARL.controlBot().getNetworkManipulator().unblockOutgoingSessionPackets();
   }
 
-  @Category(FailingTests.class)
   @Test
   public void testCreateSameFileAtOnce() throws Exception {
     ALICE.superBot().internal().createProject("foo");

--- a/travis/script/stf/master/start_stf_tests.sh
+++ b/travis/script/stf/master/start_stf_tests.sh
@@ -12,4 +12,4 @@ timeout -t 3600 ./gradlew \
   -Dstf.client.configuration.files=/home/ci/saros_src/travis/config/stf_config \
   -PskipSTFTests=false \
   -PuseBuildScan=true \
-  cleanAll :saros.stf.test:stfTest :saros.stf.test:stfFlakyAndFailingTest
+  cleanAll :saros.stf.test:stfTest :saros.stf.test:stfFlakyTest

